### PR TITLE
Adds the openvswitch dependencies, repos and optimized systemd files

### PR DESCRIPTION
**Pull request by mangelajo**
This PR adds the OpenvSwitch dependencies to MicroShift,
as well as the optimized versions of the ovs daemons
 (CPUAffinity=0 can't be injected in a different way)

Closes: https://issues.redhat.com/browse/USHIFT-141

### CODE REVIEW

File: packaging/systemd/ovs-vswitchd-micro.service [hunk 1]
```diff
@@ -0,0 +1,31 @@
+[Unit]
+Description=Open vSwitch Forwarding Unit (MicroShift)
+After=ovsdb-server.service network-pre.target systemd-udev-settle.service
+Before=network.target network.service
+Requires=ovsdb-server.service
+ReloadPropagatedFrom=ovsdb-server.service
```
Comment (zshi-redhat) [inline packaging/systemd/ovs-vswitchd-micro.service:6]: s/ovsdb-server.service/ovsdb-server-micro.service/g

Comment (mangelajo) [reply packaging/systemd/ovs-vswitchd-micro.service:6]: omg thanks a lot, I was going to test it now on RHEL4EDGE but wouldnt have workerd

File: packaging/rpm/microshift.spec [hunk 2]
```diff
@@ -152,6 +160,11 @@ install -d -m755 %{buildroot}/%{_unitdir}
 install -p -m644 packaging/systemd/microshift.service %{buildroot}%{_unitdir}/microshift.service
 install -p -m644 packaging/systemd/hostpath-provisioner.service %{buildroot}%{_unitdir}/hostpath-provisioner.service
 
+# this is temporary until we can get something equivalent from openvswitch (only adds the CPUAffinity=0)
+install -p -m644 packaging/systemd/openvswitch-micro.service %{buildroot}%{_unitdir}/openvswitch-micro.service
+install -p -m644 packaging/systemd/ovs-vswitchd-micro.service %{buildroot}%{_unitdir}/ovs-vswitchd-micro.service
+install -p -m644 packaging/systemd/ovs-vswitchd-micro.service %{buildroot}%{_unitdir}/ovsdb-server-micro.service
```
[the remark this answers is not in the document]
Comment (mangelajo) [reply packaging/rpm/microshift.spec:166]: Done

File: packaging/systemd/ovs-vswitchd-micro.service [hunk 3]
```diff
@@ -0,0 +1,31 @@
+[Unit]
+Description=Open vSwitch Forwarding Unit (MicroShift)
+After=ovsdb-server.service network-pre.target systemd-udev-settle.service
```
[the remark this answers is not in the document]
Comment (mangelajo) [reply packaging/systemd/ovs-vswitchd-micro.service:3]: Done

File: packaging/systemd/ovs-vswitchd-micro.service (hunk 1 above)
Comment (mangelajo) [reply packaging/systemd/ovs-vswitchd-micro.service:6]: Done

File: packaging/systemd/ovs-vswitchd-micro.service [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+Before=network.target network.service
+Requires=ovsdb-server.service
```
[the remark this answers is not in the document]
Comment (mangelajo) [reply packaging/systemd/ovs-vswitchd-micro.service:5]: Done

File: scripts/image-builder/build.sh [hunk 5]
```diff
@@ -158,9 +158,13 @@ rm -rf microshift-local 2>/dev/null || true
 cp -TR "${ROOTDIR}/../../packaging/rpm/_rpmbuild/RPMS" microshift-local
 createrepo microshift-local >/dev/null
 
-# Download openshift local RPM packages
+# Download openshift local RPM packages (noarch for python and selinux packages)
 rm -rf openshift-local 2>/dev/null || true
 reposync -n -a x86_64 --download-path openshift-local --repo=rhocp-4.10-for-rhel-8-x86_64-rpms >/dev/null
+reposync -n -a noarch --download-path openshift-local --repo=rhocp-4.10-for-rhel-8-x86_64-rpms >/dev/null
```
Comment (ggiguash) [inline scripts/image-builder/build.sh:164]: You should be able to provide multiple -a commands for the same repo.

**Review by ggiguash — CHANGES_REQUESTED** [review]

**Review by sallyom — CHANGES_REQUESTED** [review]
@mangelajo can you add update here? https://github.com/openshift/microshift/blob/main/docs/devenv_rhel8.md#rpm-packages  

As a follow-up (or can include in this PR - but as a follow-up I can add this):
should add a line there about installing via rpm - the next section explains installing via binary, but it's strange that we describe the rpms but don't include the `dnf install ...` cmd. 
Can add a  `Run MicroShift from rpm install` then under [`Run MicroShift Executable](https://github.com/openshift/microshift/blob/main/docs/devenv_rhel8.md#run-microshift-executable)` Can add something like "It's also possible to run MicroShift as an executable, or to replace the executable at `/usr/bin/microshift` if iterating after installing via rpm."

File: scripts/image-builder/build.sh (hunk 5 above)
Comment (mangelajo) [reply scripts/image-builder/build.sh:164]: Ah nice, will update

**Comment by mangelajo** [comment]
> @mangelajo can you add update here? https://github.com/openshift/microshift/blob/main/docs/devenv_rhel8.md#rpm-packages
> 
> As a follow-up (or can include in this PR - but as a follow-up I can add this): should add a line there about installing via rpm - the next section explains installing via binary, but it's strange that we describe the rpms but don't include the `dnf install ...` cmd. Can add a `Run MicroShift from rpm install` then under [`Run MicroShift Executable](https://github.com/openshift/microshift/blob/main/docs/devenv_rhel8.md#run-microshift-executable)` Can add something like "It's also possible to run MicroShift as an executable, or to replace the executable at `/usr/bin/microshift` if iterating after installing via rpm."

Good point!, I will update those docs.

**Comment by ggiguash** [comment]
About the dev docs update, note that I created the https://issues.redhat.com/browse/USHIFT-161 issue for me to make sure the new CSI/CNI requirements are reflected for the devenv

**Comment by mangelajo** [comment]
/retest

Enviado desde mi iPhone

File: scripts/image-builder/build.sh (hunk 5 above)
Comment (mangelajo) [reply scripts/image-builder/build.sh:164]: also multiple --repo works. Done.

**Comment by mangelajo** [comment]
@sallyom @oglok @copejon @ggiguash I've seen this passing and I added reviewer suggestions plus some missing dependencies I found. Thanks!

**Comment by mangelajo** [comment]
the rpm lock thing...

**Comment by zshi-redhat** [comment]
> /lgtm

In the case there will be other comments:

I didn't mean to trigger the merge by adding lgtm, wanted to say it looks good to me, didn't realize I have the permission to merge in this repo.

**Pull request merged**